### PR TITLE
reinstate free credits banners for Enrolled/Terminated users [risk: low]

### PR DESCRIPTION
**Pull request by davidangb**
DataBiosphere/firecloud-app#283

after discussion with PO, we are putting the free credits banners back in place for `Enrolled` and  `Terminated` statuses, and only removing/leaving removed the banner for `Enabled` status. This reverts many of the changes from #1519.

I've somewhat refactored `FreeTrialSpec` as part of this PR.

Also sneaks in a Jenkins/build fix as seen in broadinstitute/sam#287

what you see if your trial status is:

`Enrolled`:
enrolled

`Terminated`:
terminated

anything else, including blank, `Enabled`, `Finalized`:
enabled

NB: I am "implementing" the removal of the `Enabled` banner by removing its status from `trial.json`. I've done this in dev, alpha, qa, and perf; I have not done it in staging or prod but will do that as  part of the release process. Compare:

* https://storage.googleapis.com/firecloud-alerts-dev/trial.json
* https://storage.googleapis.com/firecloud-alerts/trial.json

-----

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [ ] **Submitter**: If you're adding new automated UI tests, review the test plan with QA
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Submitter**: Verify all tests go green, including CI tests and automated UI tests.
- [ ] **Submitter**: Squash commits and merge to develop. If adding test code, merge application code and test code at the same time.
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Comment by davidangb** [comment]
the single test I expected to fail did, in fact, fail. I've just edited the `trial.json` to remove `Enabled` status, which should cause the test to pass, and now:  jenkins retest.

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/user/FreeTrialSpec.scala [hunk 1]
```diff
@@ -0,0 +1,105 @@
+package org.broadinstitute.dsde.firecloud.test.user
```
Comment (davidangb) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/user/FreeTrialSpec.scala:1]: #1519 deleted this entire file; I'm now putting it back,  but have refactored it. Might be easiest to compare this to the commit prior to  #1519: https://github.com/broadinstitute/firecloud-ui/compare/a4d39cc0f197a70b87c1b731891722ab73473ca2...da_trialBannersBack

File: src/cljs/main/broadfcui/common/notifications.cljs [hunk 2]
```diff
@@ -115,6 +115,158 @@
    (fn [{:keys [state]} alert]
      (swap! state update :service-alerts #(filter (partial not= alert) %)))})
 
+(react/defc TrialAlertContainer
```
Comment (davidangb) [inline src/cljs/main/broadfcui/common/notifications.cljs:118]: #1519 deleted the `TrialAlertContainer` component; I'm now putting it back. Might be easiest to compare this to the commit prior to  #1519: https://github.com/broadinstitute/firecloud-ui/compare/a4d39cc0f197a70b87c1b731891722ab73473ca2...da_trialBannersBack

**Review by zarsky-broad — APPROVED** [review]
Compared to previous as suggested. Looks solid, didn't test.

**Pull request merged**